bundle/Controller/Admin/AdminController.php

```diff
@@ -271,10 +289,81 @@ public function handleCollectionAction(Request $request): RedirectResponse
         return $this->redirectToRoute('netgen_information_collection.route.admin.view', ['collectionId' => $collectionId]);
     }
 
+    public function editAction(Request $request, int $collectionId)
+    {
+        $this->checkEditPermissions();
+
+        $collection = $this->service->getCollection(new CollectionId($collectionId));
+
+        $locationService = $this->getRepository()->getLocationService();
+
+        $location = $locationService->loadLocation($collection->getContent()->contentInfo->mainLocationId);
+
+        $form = $this->builder->createUpdateFormForLocation($location, $collection)->getForm();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $struct = $form->getData()->payload;
+
+            $contentType = $form->getData()->definition;
+
+            $ezInfo = $this->infoCollectionRepository->find($collectionId);
```

Comment:
`$ezInfo` could be `null`, that needs to be handled.

Reply:
Also, do not shorten the variable name, it's confusing, full variant `$ezInfoCollection` is perfectly OK.

Reply:
@pspanja should I just return  the edit view if the $ezInfoCollection is null?
```
return $this->render('@NetgenInformationCollection/admin/edit.html.twig', [
                    'collection' => $collection,
                    'content' => $location->getContent(),
                    'form' => $form->createView(),
                ]);
```

Reply:
throw exception in 251ce386